app/src/main/java/io/intrepid/contest/screens/contestjudging/scoreentrydetail/expandablerecycler/ScorableCategoryViewHolder.java

```diff
@@ -0,0 +1,116 @@
+import android.widget.RatingBar;
+import android.widget.RelativeLayout;
+import android.widget.TextView;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+import io.intrepid.contest.R;
+import io.intrepid.contest.models.Category;
+
+import static android.view.View.GONE;
+import static android.view.View.VISIBLE;
+
+public class ScorableCategoryViewHolder extends RecyclerView.ViewHolder implements ExpandableViewHolder, RatingBar.OnRatingBarChangeListener {
+    private static final long COLLAPSE_DELAY = 800;
+    private final ExpandClickListener listener;
+    @BindView(R.id.category_item_title)
+    TextView categoryTitleField;
+    @BindView(R.id.expandable_relative_layout)
+    View expandableLinearLayout;
+    @BindView(R.id.editable_rating_bar_category)
+    RatingBar ratingBar;
+    @BindView(R.id.category_score_status_textview)
+    TextView scoreStatusTextView;
+
+    private Category category;
+    private boolean reveal;
+    @BindView(R.id.category_item_description)
+    TextView categoryDescriptionField;
+
+    ScorableCategoryViewHolder(ViewGroup parent, ExpandClickListener listener) {
+        super(inflateView(parent));
+        this.listener = listener;
+        ButterKnife.bind(this, itemView);
+        ratingBar.setOnRatingBarChangeListener(this);
+    }
+
+    private static View inflateView(ViewGroup parent) {
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        return inflater.inflate(R.layout.scorable_category_row_item, parent, false);
+    }
+
+    void bindCategory(Category category) {
+        bindCategory(category, true);
+    }
+
+    private void bindCategory(Category category, boolean reveal) {
+        this.category = category;
+        this.reveal = reveal;
+        categoryTitleField.setText(category.getName());
+        categoryDescriptionField.setText(category.getDescription());
+        setRatingPaneVisibility(reveal);
+    }
+
+    private void setRatingPaneVisibility(boolean reveal) {
+        this.reveal = reveal;
+        int visibility = reveal ? VISIBLE : GONE;
+        expandableLinearLayout.setVisibility(visibility);
+    }
+
+    @Override
+    public void expand() {
+        bindCategory(category, true);
+    }
+
+    @Override
+    public void collapse(){
+        bindCategory(category, false);
+    }
+
+    @Override
+    public void collapseWithDelay() {
+        new Handler().postDelayed(() -> collapse(), COLLAPSE_DELAY);
+    }
+
+    @Override
+    public boolean isExpanded() {
+        return reveal;
+    }
+
+    @Override
+    public void showTemporaryScoreMessage(@StringRes int message) {
+        scoreStatusTextView.setVisibility(VISIBLE);
+        scoreStatusTextView.setText(message);
+        new Handler().postDelayed(() -> scoreStatusTextView.setVisibility(GONE), COLLAPSE_DELAY);
+    }
+
+    @Override
+    public void toggle() {
+        if (isExpanded()) {
+            collapseWithDelay();
```

Comment:
Design said there should be a delay but did not give a specific time. I will confirm with the team.